File: 14723.py
N = int(input())

n = 1
while 1:
    an = (n ** 2 - n + 2) // 2
    if an == N: break
    elif an > N:
        n -= 1
        break

    n += 1

a = n; b = 1
an = (n ** 2 - n + 2) // 2
dis = N - an
a -= dis; b += dis

print(a, b)

File: 14723.cpp
#include <iostream>
using namespace std;

int main() {
	int N; cin >> N;

	int n = 1;
	while (1) {
		int an = (n * n - n + 2) / 2;
		if (an == N) break;
		if (an > N) {
			n--;
			break;
		}

		n++;
	}

	int a = n, b = 1;
	int an = (n * n - n + 2) / 2;
	int dis = N - an;
	a -= dis; b += dis;

	cout << a << ' ' << b << endl;
}
